src/components/dialog/dialog.scss

```diff
@@ -0,0 +1,358 @@
+    * Specifies the size of the threshold to resize the dialog.
+    * @default 4px
+    */
+  --ch-dialog-resize-threshold: 8px;
+
+  /**
+   * @prop --ch-dialog-separation-x:
+   * Specifies the separation between the action and dialog in the x axis.
+   * @default 0px
+   */
+  --ch-dialog-separation-x: 0px;
+
+  /**
+   * @prop --ch-dialog-separation-y:
+   * Specifies the separation between the action and dialog in the y axis.
+   * @default 0px
+   */
+  --ch-dialog-separation-y: 0px;
+
+  --ch-dialog-resize-threshold--half-negative: calc(
+    var(--ch-dialog-resize-threshold) * -0.5
+  );
+
+  --ch-dialog-border-inline-start-width: 0px;
+  --ch-dialog-border-inline-end-width: 0px;
+  --ch-dialog-border-block-start-width: 0px;
+  --ch-dialog-border-block-end-width: 0px;
+
+  --ch-dialog-resize__inline-start: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-inline-start-width) * 0.5
+  );
+
+  --ch-dialog-resize__inline-end: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-inline-end-width) * 0.5
+  );
+
+  --ch-dialog-resize__block-start: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-block-start-width) * 0.5
+  );
+
+  --ch-dialog-resize__block-end: calc(
+    var(--ch-dialog-resize-threshold--half-negative) -
+      var(--ch-dialog-border-block-end-width) * 0.5
+  );
+
+  --ch-dialog-dragged-x: 0px;
+  --ch-dialog-dragged-y: 0px;
+  --ch-dialog-rtl: 1;
+
+  // Necessary when the dialog is not placed in a new top layer. For example,
+  // when nesting dropdowns
+  position: fixed;
+
+  // Reset browser defaults
+  margin: 0;
+  padding: 0;
+  background-color: unset;
+  border: unset;
+  color: unset;
+  overflow: visible;
+
+  top: 0;
+  inset-inline-start: 0;
+}
+
+// - - - - - - - - - - - - - - - -
+//           Dialog
+// - - - - - - - - - - - - - - - -
+dialog {
+  // Reset browser default
+  margin: unset;
+
+  position: fixed;
+  overflow: visible;
+
+  inline-size: var(--ch-dialog-inline-size);
+  block-size: var(--ch-dialog-block-size);
+  min-inline-size: var(--ch-dialog-min-inline-size);
+  min-block-size: var(--ch-dialog-min-block-size);
+  max-inline-size: var(--ch-dialog-max-inline-size);
+  max-block-size: var(--ch-dialog-max-block-size);
+
+  //Positioning
+  inset-block-start: 0;
+  inset-inline-start: 0;
+  translate: $ch-dialog-x $ch-dialog-y;
+}
+
+// - - - - - - - - - - - - - - - -
+//             Hidden
+// - - - - - - - - - - - - - - - -
+:host([hidden]) {
+  display: block !important;
+  visibility: hidden;
+}
+
+// content-visibility: hidden is faster for rendering the content back, as
+// it preserves the rendering state of the content (display: none does not)
+@supports (content-visibility: hidden) {
+  :host([hidden]) {
+    content-visibility: hidden;
+  }
+}
+
+@supports not (content-visibility: hidden) {
+  :host([hidden]) {
+    visibility: hidden;
+  }
+}
```

Comment:
This optimization was used to not destroy the render state in nested popovers. Since we won't be showing the dialog too often and it won't be nested with other dialogs, we must remove this CSS.

```suggestion
```